Test regressions with urllib3-2.2.0: `botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes found`

After upgrading urllib3 to 2.2.0, a few tests fail (full log below):

```
FAILED tests/test_moto_api/recorder/test_recorder.py::TestThreadedMotoServer::test_server - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_or_allow_get_object[kwargs0-200-200] - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_or_allow_get_object[kwargs1-200-200] - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_or_allow_get_object[kwargs2-200-403] - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_or_allow_get_object[kwargs3-200-200] - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_or_allow_get_object[kwargs4-403-403] - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_or_allow_get_object[kwargs5-200-403] - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_put_object - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_all_actions - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
ERROR tests/test_s3/test_s3_bucket_policy.py::TestBucketPolicy::test_block_all_with_different_principal - botocore.exceptions.HTTPClientError: An HTTP Client raised an unhandled exception: sequence item 0: expected str instance, bytes f...
```

To reproduce:
```
pip install urllib3==2.2.0
python -m pytest tests/test_s3/test_s3_bucket_policy.py tests/test_moto_api/recorder/test_recorder.py
```

Full log (400k):  [log.txt](https://github.com/getmoto/moto/files/14166999/log.txt)

--- a/.github/workflows/test_outdated_versions.yml
+++ b/.github/workflows/test_outdated_versions.yml
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        botocore: ["--upgrade boto3 botocore", "boto3==1.11.0 botocore==1.14.0"]
+        botocore: ["--upgrade boto3 botocore", "--upgrade boto3 botocore && pip install urllib3==2.2.1", "boto3==1.11.0 botocore==1.14.0"]
         python-version: [ "3.11" ]
         responses-version: ["0.15.0", "0.17.0", "0.19.0", "0.20.0" ]
         werkzeug-version: ["2.0.1", "2.1.1", "2.2.2"]

